airflow/providers/amazon/aws/operators/emr.py

```diff
@@ -830,11 +849,32 @@ def execute(self, context: Context) -> None:
         self.log.info("Terminating JobFlow %s", self.job_flow_id)
         response = emr.terminate_job_flows(JobFlowIds=[self.job_flow_id])
 
+        if self.deferrable:
```

Comment:
oh we're adding deferrable capability to an operator that didn't have a `wait_for_completion` ?

Reply:
Yea, deferring essentially enforces a `wait_for_completion`. It makes sense to add that functionality in for the synchronous path as well. 
See [this comment](https://github.com/apache/airflow/pull/30864#discussion_r1178385960) in a previous PR

Reply:
Yea, I don't think there will be any issues with doing something like this.